src/interviewkit/slicer.py

```diff
@@ -18,102 +18,90 @@
-1hr 30mins 40sec = 90mins 40secs
-
 python interviewkit/slicer.py data/Martine+Barrat_FINAL.mp3 80:30 90:40
 
 """
 import sys
 from pathlib import Path
 import shutil
+from datetime import timedelta, datetime
+from dataclasses import dataclass
+import pydub
 
-try:
-    import pydub
-except ImportError:
-    print("Please install pydub: pip install pydub")
-    exit(1)
 
 if shutil.which("ffmpeg") is None:
     print("Please install ffmpeg: https://ffmpeg.org/download.html")
     print("  On mac you can: brew install ffmpeg")
     exit(1)
 
-def convert_audio_time_to_msec(audio_time_split_list):
-    """ Converting mins and secs to msecs for pydub computation """
-
-    if(audio_time_split_list):
-        if(len(audio_time_split_list) == 1):
-            return int(audio_time_split_list[0]) * 60 * 1000
-        elif(len(audio_time_split_list) == 2):
-            return int(audio_time_split_list[0]) * 60 * 1000 + int(audio_time_split_list[1]) * 1000
-        else:
-            print("Error! Audio slice input params invalid. Audio slice supports start/end time in mins or mins:secs format. Please try again with correct input times.")
-            print("Error inside convert_audio_time_to_msec(audio_time_split_list) funtion.")
-            exit(1)
-    else:
-        print("Error! Audio slice input params invalid. Please try again with correct parameters.")
-        print("Error inside convert_audio_time_to_msec(audio_time_split_list) funtion.")
-        exit(1)
-
-def export_filename(audio_time_list):
-    """ Filename for exported file """
-    
-    if audio_time_list and len(audio_time_list) == 2:
-        return f"{audio_time_list[0]}m{audio_time_list[1]}s"
-    elif audio_time_list and len(audio_time_list) == 1:
-        return f"{audio_time_list[0]}m"
-    else:
-        print("Error! Audio slice input params invalid. Please try again with correct parameters.")
-        print("Error inside export_filename(audio_time_list) funtion.")
-        exit(1)
-
-
-def audio_slicing(path, audio_slice_start_time, audio_slice_end_time):
-    """ It reads the original audio and uses start and end input time params to generate sliced audio. """
-    
-    print("Sampling {} from {} to {}".format(path, audio_slice_start_time, audio_slice_end_time))
 
-    # Reading original audio file
-    audio = pydub.AudioSegment.from_file(path)
-    original_audio_size_ms = audio.duration_seconds * 1000
+EXPECTED_TIME_FORMAT = "%H:%M:%S"
 
-    # Fetching mins and secs from audio input
-    audio_start_time_list = audio_slice_start_time.split(":")
-    audio_end_time_list = audio_slice_end_time.split(":")
+@dataclass
+class SlicerInput:
+    path: Path
+    start_time: timedelta
+    end_time: timedelta
 
-    # Converting audio start and end times in msecs
-    audio_start_time = convert_audio_time_to_msec(audio_start_time_list)
-    audio_end_time = convert_audio_time_to_msec(audio_end_time_list)
+def convert_time_input_to_time_delta(time_input: str):
```

Comment:
I think your instinct to reuse something like timedelta here was great. I empathize with the challenges you ran into: parsing time strings is not easy! 

I can see how you came to the conclusion. You might need a bespoke parser. I feel like someone must have written a moment.js equivalent for Python by now. Maybe there is something like that.

Reply:
Thanks Audrey. Yes it started out quite clean, and then slowly started getting less and less haha. Unfortunate as it could have been a fairly simple solution. It's quite surprising a time concept on its own doesn't really exist in the std library, only when associated with date time. Ah nice, I'll see if there's a library about, I feel like there must be. 

Thanks for your review, massively appreciate it